### PR TITLE
add composer, gh workflows and vue8 support

### PR DESCRIPTION
**Pull request by kyteinsky**
todo: vue8

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1 +1 @@
-/appinfo/info.xml    @julien-nc
+/appinfo/info.xml    @julien-nc @kyteinsky
```
Comment (julien-nc) [inline .github/CODEOWNERS:1]:
```suggestion
/appinfo/info.xml    @kyteinsky @julien-nc
```

File: composer.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    {
+			"name": "Julien Veyssier",
+			"email": "julien-nc@posteo.net"
+		}
+	],
+	"require": {
+		"php": "^8.0 || ^8.1 || ^8.2",
+		"bamarni/composer-bin-plugin": "^1.8"
+	},
+	"require-dev": {
```
Comment (julien-nc) [inline composer.json:16]: Why not adding all dependencies here and having them in vendor-bin?

**Review by julien-nc — CHANGES_REQUESTED** [review]
LGTM

File: composer.json (hunk 2 above)
Comment (kyteinsky) [reply composer.json:16]: what do you mean? Moving all vendor-bin deps in composer.json directly without the bin support? I think this is a good insurance for the future if anything decides to conflict the core deps.

File: .github/CODEOWNERS (hunk 1 above)
Comment (kyteinsky) [reply .github/CODEOWNERS:1]: > If you want to match two or more code owners with the same pattern, all the code owners must be on the same line. If the code owners are not on the same line, the pattern matches only the last mentioned code owner.

All have the same priority. Why is it limited to only info.xml though?

File: composer.json (hunk 2 above)
Comment (julien-nc) [reply composer.json:16]: Nevermind, my question was badly formulated.

**Review by julien-nc — APPROVED** [review]

**Comment by kyteinsky** [comment]
screenshots below. There is a slight glitch with the selector dropdown but not an issue from this end and did auto-correct itself.

Screenshot from 2024-03-06 10-14-07
Screenshot from 2024-03-06 10-14-17
Screenshot from 2024-03-06 10-14-36
Screenshot from 2024-03-06 10-14-45

File: .github/CODEOWNERS (hunk 1 above)
Comment (marcelklehr) [reply .github/CODEOWNERS:1]: This is Nextcloud specific syntax, for the security bulletin done internally by the security team

**Review by marcelklehr — APPROVED** [review]

**Pull request merged**